Fix a11y layout issues

mtwente opened the pull request:
# Pull request

## Proposed changes

* fix #92 
  * improve colour contrasts
  * translate aria-labels to German
  * add missing header levels
  * various small styling improvements

## Types of changes

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] New feature (non-breaking change which adds functionality).
- [x] Enhancement (non-breaking change which enhances functionality)
- [x] Bug Fix (non-breaking change which fixes an issue).
- [ ] Breaking change (fix or feature that would cause existing functionality to change).

## Checklist

<!-- Go over all the following points, and put an `x` in all the boxes that apply. If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have read the **[CONTRIBUTING](./CONTRIBUTING.md)** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Enhanced export functionality in data tables with new button configurations and a modal display on click.
- **Style**
	- Updated button styles across multiple pages for a more cohesive look.
	- Improved accessibility by updating aria-label attributes and heading levels for better user experience.
- **Bug Fixes**
	- Removed redundant includes to streamline HTML files.
- **Documentation**
	- Updated labels on download buttons for better clarity and consistency.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

mtwente commented on _sass/_custom.scss:
```diff
@@ -70,10 +67,80 @@
 }
 
 a {
+    &:hover {
+        color: #191919;
+    }
```
That's OK since there is a separate class for links on `#3a1e3e` background that meets the minimum requirement.

koilebeit reviewed (commented):
I like it! Thank you for your great work!

Two three questions/comments, mainly about the buttons:
- Is there a reason why the buttons are sometimes yellow, but mostly white with purple writing? (e.g. the sort buttons on the browse page and on the topics page). Should we perhaps make all the buttons the same?
- Accordingly, would it perhaps be more intuitive if the eras in the wordcloud were yellow, like the tags on the objects?
- If the buttons on the metadata page (Excel, CSV) remain yellow, would it be possible to separate them visually?
- In the metadata table, the selection of rows to be displayed is still blue.

mtwente commented:
> I like it! Thank you for your great work!
> 
> Two three questions/comments, mainly about the buttons:
> 
> * Is there a reason why the buttons are sometimes yellow, but mostly white with purple writing? (e.g. the sort buttons on the browse page and on the topics page). Should we perhaps make all the buttons the same?

The initial idea was to have the buttons that you would use to access anything related to the objects' _content_ (era/content tags, search for objects) in secondary/yellow and the buttons that you would use to access any other information, mainly _"technical"/format_ (meta)data, in outline-primary/purple (file formats, download options, displaying/file ordering on the website).

As far as I see, this works with the exception of the two download buttons on the data table page (the wordcloud I already fixed, see below).

> * Accordingly, would it perhaps be more intuitive if the eras in the wordcloud were yellow, like the tags on the objects?

Absolutely, I changed the buttons on the cloud page to secondary/yellow.

> * If the buttons on the metadata page (Excel, CSV) remain yellow, would it be possible to separate them visually?

I do not like the button setup on that page anyway. There are these two buttons for filtered downloads and then, at the top right of the page, another download button for the unfiltered data. I would like to have them in one place and (also in the current setup) the button for unfiltered download needs a better label/description, e.g. "Kompletten Datensatz herunterladen" or "weitere Formate (ungefiltert)" etc.

> * In the metadata table, the selection of rows to be displayed is still blue.

Yes, that's because it's not styled via the bootstrap options, but comes from DataTables directly

mtwente commented:
@koilebeit I rearranged the export/download buttons on the data page. I'm not sure if the button labels are clear in terms of what they are for.
<img width="659" alt="data_new_buttons" src="https://github.com/Stadt-Geschichte-Basel/forschung.stadtgeschichtebasel.ch/assets/127845092/d4f031f0-3dea-48a5-9f85-62063ca7a53c">

koilebeit reviewed (approved):
Now it looks great! Thanks!
